src/main/java/org/quantumbadger/redreader/http/okhttp/OKHTTPBackend.java

```diff
@@ -31,11 +35,26 @@
 public class OKHTTPBackend implements HTTPBackend {
 
 	private final OkHttpClient mClient;
+	private static HTTPBackend httpBackend;
+	public static boolean reset = false;
+
+	public static HTTPBackend getHttpBackend(boolean force) {
+		if(force || httpBackend == null) {
+			httpBackend = new OKHTTPBackend();
+			reset = false;
+		}
+		return httpBackend;
+	}
 
 	public OKHTTPBackend() {
 
 		mClient = new OkHttpClient();
 
+		if(BaseActivity.getTorStatus()) {
+			Proxy tor = new Proxy(Proxy.Type.HTTP, new InetSocketAddress("127.0.0.1", 8118));
```

Comment:
Oh, and, HTTP proxies handle HTTPS just fine. If the IP address is something other than 127.0.0.1, then this could be a security issue, since the browser would transmit in plain text to the proxy.